Add data licensing blog (#314)
epiverse-trace/epiverse-trace.github.io: posts/data-licensing.qmd

format:
  html: 
    toc: true
# doi: "10.59350/z78kb-qrz59"
---

Licenses are an important topic within open source. Without licenses, information or code can be publicly available but not legally available for reuse or redistribution. The open source software community's most common licenses are [the MIT license](https://mit-license.org) or the [GNU GPLv3](https://www.gnu.org/licenses/gpl-3.0.html).

When you read the MIT or GNU license, you can see they are rather specific:

> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”) [MIT License]

and

> The GNU General Public License is a free, copyleft license for software and other kinds of works.

They aim to cover primarily software, not other forms of information such as, for example, data.

### Licensing only code in R packages

Given the importance of a license for redistribution, CRAN requires R packages to contain an [open-source license](https://cran.r-project.org/doc/FAQ/R-exts.html#Licensing).[^1] As one of the most important distributors for R packages historically, this rule has become the de facto standard.

> An R package is licensed appropriately when the license is stated in both the metadata `DESCRIPTION` file and the `LICENSE` file. The `LICENSE` file should include the license year and exact copyright holder (for example, an organization or a specific individual). Some R package developers choose to include a copy of the license but this is [not bundled in the R package when it is built](https://r-pkgs.org/license.html#key-files). You can [find an example of a `DESCRIPTION` file here](https://github.com/epiverse-trace/cfr/blob/29ee12aa9b8a49eedb9207d95c25c13d7d4e0ace/DESCRIPTION#L27) and [find an example `LICENSE` here](https://github.com/epiverse-trace/cfr/blob/main/LICENSE).

Reviewer: This is a nitpick and maybe not worth the complication but: not all licenses require a `LICENSE` file. If the package is licensed GPL for example, just stating it in the `DESCRIPTION` is enough.

`LICENSE` is required for MIT and non-standard licenses

Author: ```suggestion
An R package is licensed appropriately when the license is stated in the metadata `DESCRIPTION` file and, if necessary, in the `LICENSE` file. We recommend including the `LICENSE` file, with the license year and exact copyright holder (for example, an organization or a specific individual). Some R package developers choose to include a copy of the license but this is [not bundled in the R package when it is built](https://r-pkgs.org/license.html#key-files). You can [find an example of a `DESCRIPTION` file here](https://github.com/epiverse-trace/cfr/blob/29ee12aa9b8a49eedb9207d95c25c13d7d4e0ace/DESCRIPTION#L27) and [find an example `LICENSE` here](https://github.com/epiverse-trace/cfr/blob/main/LICENSE).
```

Like this maybe?

Data need licenses such as the Public Domain Dedication or the Creative Commons Attribution to maximize redistribution. Such [general licenses](https://r-pkgs.org/license.html#code-you-write) help minimize differences among countries that do and do not recognize database rights. Given that any copyright limitations (even attribution) mean they only apply when the database right is recognized, we recommend the most minimal option: The Public Domain Dedication. This levels the playing field for reuse and redistribution, no matter the jurisdiction.

### Licensing code and data in one R package

But what to do if your R package has both code and data as primary objects of (roughly) equal importance?

In these cases a software license inadequately covers the data, and a data license inadequately covers the code. Dual licensing can help resolve this issue. This means there is one license for code (for example, MIT license)  and another license for the included data (for example, Public Domain Dedication).

> After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, in our own [**epiparameter** package](https://github.com/epiverse-trace/epiparameter) we dual licensed the code and data as well. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/4f7008063040fc4a0b436a9e4d2adb9bb7022f6c/DESCRIPTION#L21) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/main/LICENSE) to license the data under CC0. When including data in your R package from other sources it is important to check that the license of your package and the data is compatible[^4], or that the individual data license is clearly stated, as in igrphdata.

Reviewer: ```suggestion
After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, in our own [**epiparameter** package](https://github.com/epiverse-trace/epiparameter) we dual licensed the code and data as well. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/4f7008063040fc4a0b436a9e4d2adb9bb7022f6c/DESCRIPTION#L21) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/main/LICENSE) to license the data under CC0. When including data in your R package from other sources it is important to check that the license of your package and the data is compatible[^4], or that the individual data license is clearly stated, as in igraphdata.
```

> After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, in our own [**epiparameter** package](https://github.com/epiverse-trace/epiparameter) we dual licensed the code and data as well. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/4f7008063040fc4a0b436a9e4d2adb9bb7022f6c/DESCRIPTION#L21) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/main/LICENSE) to license the data under CC0. When including data in your R package from other sources it is important to check that the license of your package and the data is compatible[^4], or that the individual data license is clearly stated, as in igraphdata.

Author: ```suggestion
After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, in our own [**epiparameter** package](https://github.com/epiverse-trace/epiparameter) we dual licensed the code and data as well. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/4f7008063040fc4a0b436a9e4d2adb9bb7022f6c/DESCRIPTION#L21) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/main/LICENSE) to license the data under CC0.  Concretely, we include this additional text  in `LICENSE` to clarify the dual license and that we recommend citing the original source regardless:

>  All data included in the epiparameter R package is licensed under CC0 (https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt). This includes the parameter database (extdata/parameters.json) and data in the data/ folder. Please cite the individual parameter entries in the database when used.

When including data in your R package from other sources it is important to check that the license of your package and the data is compatible[^4], or that the individual data license is clearly stated, as in igraphdata.
```

@joshwlambert what do you think of this formulation based on @adamkucharski's comment?

Reviewer 2: I think the new text is good, but I don't think the original text should be removed. I would just add it after "... under CC0." and leave the sentence "When including data in your R package ..." at the end of the paragraph. 

I think another sentence is also required to fully address Adam's point. We need to clarify that the model estimates are treated an non-copyrightable facts. This could either be added to the paragraph on L39 or this paragraph and make specific reference to {epiparameter}.

Author: The original text was only moved lower 😊 Here's a revised version based on your feedback. 👇 

```suggestion
After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, in our own [**epiparameter** package](https://github.com/epiverse-trace/epiparameter) we dual licensed the code and data as well. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/4f7008063040fc4a0b436a9e4d2adb9bb7022f6c/DESCRIPTION#L21) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/main/LICENSE) to license the data under CC0.  Concretely, we include this additional text  in `LICENSE` to clarify the dual license and that we recommend citing the original source regardless:

>  All data included in the epiparameter R package is licensed under CC0 (https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt). This includes the parameter database (extdata/parameters.json) and data in the data/ folder. Please cite the individual parameter entries in the database when used.

When including data in your R package from other sources it is important to check that the license of your package and the data is compatible[^4], or that the individual data license is clearly stated, as in igraphdata. For epiparameter, we consider model estimates as facts (that is, not copyrightable).
```

@joshwlambert - could you confirm the final sentence is as you expected? If you have concrete suggestions, as you do here, it helps if you suggest them directly so there is no loss during my interpretation of your suggestion.

In these cases a software license inadequately covers the data, and a data license inadequately covers the code. Dual licensing can help resolve this issue. This means there is one license for code (for example, MIT license)  and another license for the included data (for example, Public Domain Dedication).

After conducting an online search, dual licensing for R packages seems rare. An interesting example of dual licensing is the [**igraphdata** package](https://github.com/igraph/igraphdata/blob/main/LICENSE), which contains several licenses: One for each dataset included in the package. Similar to igraphdata, in our own [**epiparameter** package](https://github.com/epiverse-trace/epiparameter) we dual licensed the code and data as well. We [licensed the code using the `DESCRIPTION` file](https://github.com/epiverse-trace/epiparameter/blob/4f7008063040fc4a0b436a9e4d2adb9bb7022f6c/DESCRIPTION#L21) and used [the `LICENSE` file](https://github.com/epiverse-trace/epiparameter/blob/main/LICENSE) to license the data under CC0.  Concretely, we include this additional text  in `LICENSE` to clarify the dual license and that we recommend citing the original source regardless:

> > All data included in the epiparameter R package is licensed under CC0 (https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt). This includes the parameter database (extdata/parameters.json) and data in the data/ folder. Please cite the individual parameter entries in the database when used.

Author: ```suggestion
> All data included in the epiparameter R package is licensed under CC0 (<https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt>). This includes the parameter database (extdata/parameters.json) and data in the data/ folder. Please cite the individual parameter entries in the database when used.
```
